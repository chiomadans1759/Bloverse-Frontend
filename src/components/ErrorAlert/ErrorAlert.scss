#error-alert {
  .error {
    border-radius: 0;
    background: #FF2737;
    color: #ffffff;
    font-weight: 600;
    border: 0.063rem solid rgba(0, 0, 0, 0.15);
    padding: 1.5rem;
  }
  .error-icon {
    color: #ffffff;
    position: absolute;
    right: .5rem;
    top: .5rem;
  }
}
