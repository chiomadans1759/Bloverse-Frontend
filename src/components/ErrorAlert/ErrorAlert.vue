<template>
  <main id="error-alert">
    <div class="alert alert-danger error" role="alert">
      <span class="d-flex justify-content-end">
        <i class="fal fa-times error-icon"></i>
      </span>
      <h4 v-if="heading" class="alert-heading" >{{heading}}</h4>
      <p class="text"><slot></slot></p>
    </div>
  </main>
</template>

<script>
export default {
  name:'Error',
  props: {
    heading: {
      type: String,
      required: false,
      default: ""
    }
  }

}
</script>

<style lang="scss" scoped>
@import './ErrorAlert'
</style>

