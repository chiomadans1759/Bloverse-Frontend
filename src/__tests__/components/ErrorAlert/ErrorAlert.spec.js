import { mount } from '@vue/test-utils';
import ErrorAlert from '@/components/ErrorAlert/ErrorAlert.vue';

describe('ErrorAlert Component', () => {
  it('should render correctly', () => {
    const wrapper = mount(ErrorAlert);

    expect(wrapper.element).toMatchSnapshot();
  });
  it('should render title', () => {
    const wrapper = mount(ErrorAlert, {
      propsData: {
        heading: 'alert-heading'
      }
    })
    expect(wrapper.props().heading).toBe('alert-heading');
  })
});
