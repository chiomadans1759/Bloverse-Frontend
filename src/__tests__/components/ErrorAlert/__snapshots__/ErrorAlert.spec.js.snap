// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert Component should render correctly 1`] = `
<main
  id="error-alert"
>
  <div
    class="alert alert-danger error"
    role="alert"
  >
    <span
      class="d-flex justify-content-end"
    >
      <i
        class="fal fa-times error-icon"
      />
    </span>
     
    <!---->
     
    <p />
  </div>
</main>
`;
